README.md

```diff
@@ -1,3 +1,51 @@
 # Silksong.ModMenu
 
-A Hollow Knight: Silksong mod.
+The standard menu-modification mod for Silksong. It provides a small but powerful set of features for mods to use.
+
+This mod also adds a 'Mod Options' menu to the main Options menu by default.
+
+The rest of this page is targeted towards mod developers. If you're not developing mods, you can stop reading now.
+
+## Mod Options
+
+This mod adds a 'Mods' sub-menu to the bottom of the Options menu as its main feature. All other plugins can add content to this menu as they wish.
+
+By default, any installed plugin's [`ConfigFile`](https://docs.bepinex.dev/master/api/BepInEx.Configuration.ConfigFile.html) data will be used to generate an ingame menu that can modify those configuration options. At this time, only booleans, enums, and explicit value lists are supported; for more complex configuration including arbitrary integers and floats, manual menu construction is needed.
+
+Plugins can add aub-menus to the main menu through the [`Registry`](Registry.cs) class manually, or by implementing one of the [Plugin](Plugin) interfaces that extend from [`IModMenuInterface`](Plugin/IModMenuInterface.cs) for a simplified workflow. Plugins can add the [`ModMenuIgnore`](Plugin/ModMenuIgnore.cs) attribute to opt out of automatic menu creation.
+
+## New Game Types
+
+ModMenu also modifies the "New Game" menu, allowing the addition of new game modes, such as modded playthroughs, plandos, and Randomizer runs.
```

Comment:
Maybe the readme should indicate that this feature is planned (and doesn't exist). (I believe ModMenu ought to be released before adding a fully-featured new game mode API)